src/util.cpp

```diff
@@ -667,76 +667,118 @@ void CLanguageComboBox::OnLanguageActivated ( int iLanguageIdx )
+
+    // qInfo() << qUtf8Printable ( QString ( "Parsing network address %1" ).arg ( strAddress ) );
 
     // init requested host address with invalid address first
     HostAddress = CHostAddress();
 
-    // parse input address for the type "IP4 address:port number" or
-    // "[IP6 address]:port number" assuming the syntax is correctly given
-    QStringList slAddress = strAddress.split ( ":" );
-    QString     strSep    = ":";
-    bool        bIsIP6    = false;
+    // Allow the following address formats:
+    // [addr4or6]
+    // [addr4or6]:port
+    // addr4
+    // addr4:port
+    // hostname
+    // hostname:port
+    // (where addr4or6 is a literal IPv4 or IPv6 address, and addr4 is a literal IPv4 address
+
+    bool    bLiteralAddr = false;
+    QRegExp rx1 ( "^\\[([^]]*)\\](?::(\\d+))?$" ); // [addr4or6] or [addr4or6]:port
+    QRegExp rx2 ( "^([^:]*)(?::(\\d+))?$" );       // addr4 or addr4:port or host or host:port
+
+    QString strPort;
 
-    if ( slAddress.count() > 2 )
+    // parse input address with rx1 and rx2 in turn, capturing address/host and port
+    if ( rx1.indexIn ( strAddress ) == 0 )
     {
-        // IP6 address
-        bIsIP6 = true;
-        strSep = "]:";
+        // literal address within []
+        strAddress   = rx1.cap ( 1 );
+        strPort      = rx1.cap ( 2 );
+        bLiteralAddr = true; // don't allow hostname within []
+    }
+    else if ( rx2.indexIn ( strAddress ) == 0 )
+    {
+        // hostname or IPv4 address
+        strAddress = rx2.cap ( 1 );
+        strPort    = rx2.cap ( 2 );
+    }
+    else
+    {
+        // invalid format
+        // qInfo() << qUtf8Printable ( QString ( "Invalid address format" ) );
+        return false;
     }
-
-    QString strPort = strAddress.section ( strSep, 1, 1 );
 
     if ( !strPort.isEmpty() )
     {
-        // a colon is present in the address string, try to extract port number
+        // a port number was given: extract port number
         iNetPort = strPort.toInt();
 
-        // extract address port before separator (should be actual internet address)
-        strAddress = strAddress.section ( strSep, 0, 0 );
-
-        if ( bIsIP6 )
+        if ( iNetPort >= 65536 )
         {
-            // remove "[" at the beginning
-            strAddress.remove ( 0, 1 );
+            // invalid port number
+            // qInfo() << qUtf8Printable ( QString ( "Invalid port number specified" ) );
+            return false;
         }
     }
 
     // first try if this is an IP number an can directly applied to QHostAddress
-    if ( !InetAddr.setAddress ( strAddress ) )
+    if ( InetAddr.setAddress ( strAddress ) )
     {
-        // it was no valid IP address, try to get host by name, assuming
+        if ( !bEnableIPv6 && InetAddr.protocol() == QAbstractSocket::IPv6Protocol )
+        {
+            // do not allow IPv6 addresses if not enabled
+            // qInfo() << qUtf8Printable ( QString ( "IPv6 addresses disabled" ) );
```

Comment:
Do we give an error message to the user somewhere else?

Reply:
At the moment, with all versions, if an invalid hostname or IP address is entered into the Connect window, the connection fails silently. It would be good to fix that more generally, but I think that should be separate from this PR. Maybe raise it as a specific issue?

Reply:
Done #1945